eth/state_encoder.go

```diff
@@ -59,7 +59,7 @@ type keEthAddr struct{}
 
 func (_ keEthAddr) Encode(value gethcommon.Address) []byte { return value.Bytes() }
 func (_ keEthAddr) Decode(bz []byte) (int, gethcommon.Address) {
-	return gethcommon.AddressLength, gethcommon.BytesToAddress(bz)
+	return gethcommon.AddressLength, gethcommon.BytesToAddress(bz[:gethcommon.AddressLength])
```

Comment:
This was the bug. `bz` was coming in as the entire PairKey (address + storage slot key), and `gethcommon.BytesToAddress` was only taking the right-most 20 bytes, whereas it should be the left-most 20 bytes.